tests/testdsl/testdsl.go

```diff
@@ -141,6 +142,84 @@ func RequireOutputWithin(output string, within time.Duration) Proc {
 	}
 }
 
+func RequireStepRetries(step string, count int) Proc {
+	return func(ctx context.Context, td *TestData) error {
+		var backoffTime uint
+
+		for i := 0; i < count; i++ {
+			fmt.Printf("> Checking step %s performs retry %d of %d\n", step, i+1, count)
+
+			backoffTime = uint(10) << i
+
+			fmt.Printf("\t> Checking attempt #%d executes (waiting %d seconds)\n", i+1, backoffTime)
+			if err := timeout(time.Second*time.Duration(backoffTime), func() error {
+				return requireLogFields(ctx, td, map[string]any{
+					"caller":  "executor",
+					"step":    step,
+					"message": "executing step",
+					"attempt": i,
+				})
+			}); err != nil {
+				return err
+			}
+
+			if i+1 < count {
```

Comment:
Maybe revert this to lower the indent count?  I typically prefer to guard data patterns then quit, vs do things in if:

```
if i+1 >= count {
  continue
}
// ...do logic here if i < count

Reply:
Yep yep big agree.